### PR TITLE
Non-GPU environment for publishing docs

### PR DESCRIPTION
**Pull request by eightysteele**
This PR introduces a new `docs` environment to get `pdoc` rendering on non-GPU runners. Essentially it pulls in CPU-capable versions of `jaxlib` and friends while rendering. This is needed because `pdoc` uses dynamic analysis to render docstrings, meaning that it literally imports and executes the code when it runs. There's one module with `assert` statements that kills the render process (see [comment](https://github.com/probcomp/b3d/pull/106#issuecomment-2285031902)), so that's commented out for now and I captured it as an issue [here](https://linear.app/chi-fro/issue/GEN-451/remove-assert-statements-in-other-likelihoodspy).

### CODE REVIEW

**Comment by eightysteele** [comment]
@georgematheos to get this PR working I needed to comment out the `assert` statements in [src/b3d/chisight/dense/likelihoods/other_likelihoods.py](https://github.com/probcomp/b3d/pull/106/files#diff-e7cd3b007f77b9124be9125aef754b3279e506a546b21524488049f1105eb650). If possible, could we potentially get those into a test file and do assertions through `pytest`? In the meantime, docs are now publishing 💥 [https://b3d.gen.dev](https://b3d.gen.dev)

File: pyproject.toml [hunk 1]
```diff
@@ -34,13 +34,17 @@ rerun-sdk = "==0.17.0"
...
+doc = { features = ["no-cuda", "core", "gl"], solve-group = "no-cuda" }
+gpu = { features = ["cuda", "core", "gl"], solve-group = "cuda" }
 
 [tool.pixi.activation.env]
 XLA_PYTHON_CLIENT_PREALLOCATE = "false"
 XLA_PYTHON_CLIENT_ALLOCATOR = "platform"
 CPLUS_INCLUDE_PATH = "$CONDA_PREFIX/targets/x86_64-linux/include"
 
+[tool.pixi.feature.no-cuda.activation.env]
+LD_LIBRARY_PATH = "$CONDA_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib64"
```
Comment (eightysteele) [inline pyproject.toml:46]: needed for the `opengl` shared libraries.

File: pyproject.toml [hunk 2]
```diff
@@ -34,13 +34,17 @@ rerun-sdk = "==0.17.0"
 
 [tool.pixi.environments]
 cpu = { features = [], solve-group = "default" }
-gpu = { features = ["core", "cuda", "gl"], solve-group = "cuda" }
+doc = { features = ["no-cuda", "core", "gl"], solve-group = "no-cuda" }
```
Comment (eightysteele) [inline pyproject.toml:37]: here's the new `docs` environment.

File: pyproject.toml [hunk 3]
```diff
@@ -124,17 +133,30 @@ description = "sync test data"
...
-cmd = "python -m pdoc --logo https://github.com/probcomp/b3d/assets/66085644/50bc2fc3-c9cd-4139-bed6-9c6d53933622 --math --mermaid -o docs/static/ b3d"
-env = { PDOC_ALLOW_EXEC = "1" }
-
 [tool.pixi.pypi-options]
 index-url = "https://pypi.org/simple"
 extra-index-urls = [
     "https://oauth2accesstoken@us-west1-python.pkg.dev/probcomp-caliban/probcomp/simple",
 ]
 
+[tool.pixi.feature.no-cuda]
```
Comment (eightysteele) [inline pyproject.toml:142]: the `no-cuda` features just pulls in CPU-capable versions of `jaxlib`, `pytorch`, etc.

**Comment by georgematheos** [comment]
@eightysteele this looks good to me.  Agreed that the right move is to factor those assert statements into a pytest test file.  I will make an issue for adding back in the assert tests using pytest, and plan to circle back to it next time I am touching that part of the codebase.  Feel free to merge these changes with the commented out asserts (or feel free to remove the asserts fully if you prefer).

**Review by georgematheos — APPROVED** [review]

**Comment by eightysteele** [comment]
@georgematheos sounds good. i removed the `assert` statements. if helpful, i captured issue `GEN-451` (but feel free to mark that as a duplicate if you already created one).

**Pull request merged**